gcsfs/core.py

```diff
@@ -1294,7 +1294,7 @@ async def _find(
 
         if maxdepth:
             # Filter returned objects based on requested maxdepth
-            depth = path.count("/") + maxdepth
+            depth = path.rstrip("/").count("/") + maxdepth
```

Comment:
The depth is wrong when there is a trailing slash in the source path. We could also strip it at the very beginning of the `_find` method (the tests also works).